src/um/winspool.rs

```diff
@@ -36,3 +2146,283 @@ STRUCT!{struct PRINTER_OPTIONSW {
+    ftDriverDate: FILETIME,
+    dwlDriverVersion: DWORDLONG,
+    szPackageID: [WCHAR; MAX_PATH],
+}}
+pub type PCORE_PRINTER_DRIVERW = *mut CORE_PRINTER_DRIVERW;
+extern "system" {
+    pub fn GetCorePrinterDriversA(
+        pszServer: LPCSTR,
+        pszEnvironment: LPCSTR,
+        pszzCoreDriverDependencies: LPCSTR,
+        cCorePrinterDrivers: DWORD,
+        pCorePrinterDrivers: PCORE_PRINTER_DRIVERA,
+    ) -> HRESULT;
+    pub fn GetCorePrinterDriversW(
+        pszServer: LPCWSTR,
+        pszEnvironment: LPCWSTR,
+        pszzCoreDriverDependencies: LPCWSTR,
+        cCorePrinterDrivers: DWORD,
+        pCorePrinterDrivers: PCORE_PRINTER_DRIVERW,
+    ) -> HRESULT;
+    pub fn CorePrinterDriverInstalledA(
+        pszServer: LPCSTR,
+        pszEnvironment: LPCSTR,
+        CoreDriverGUID: GUID,
+        ftDriverDate: FILETIME,
+        dwlDriverVersion: DWORDLONG,
+        pbDriverInstalled: *mut BOOL,
+    ) -> HRESULT;
+    pub fn CorePrinterDriverInstalledW(
+        pszServer: LPCWSTR,
+        pszEnvironment: LPCWSTR,
+        CoreDriverGUID: GUID,
+        ftDriverDate: FILETIME,
+        dwlDriverVersion: DWORDLONG,
+        pbDriverInstalled: *mut BOOL,
+    ) -> HRESULT;
+    pub fn GetPrinterDriverPackagePathA(
+        pszServer: LPCSTR,
+        pszEnvironment: LPCSTR,
+        pszLanguage: LPCSTR,
+        pszPackageID: LPCSTR,
+        pszDriverPackageCab: LPSTR,
+        cchDriverPackageCab: DWORD,
+        pcchRequiredSize: LPDWORD,
+    ) -> HRESULT;
+    pub fn GetPrinterDriverPackagePathW(
+        pszServer: LPCWSTR,
+        pszEnvironment: LPCWSTR,
+        pszLanguage: LPCWSTR,
+        pszPackageID: LPCWSTR,
+        pszDriverPackageCab: LPWSTR,
+        cchDriverPackageCab: DWORD,
+        pcchRequiredSize: LPDWORD,
+    ) -> HRESULT;
+    pub fn DeletePrinterDriverPackageA(
+        pszServer: LPCSTR,
+        pszInfPath: LPCSTR,
+        pszEnvironment: LPCSTR,
+    ) -> HRESULT;
+    pub fn DeletePrinterDriverPackageW(
+        pszServer: LPCWSTR,
+        pszInfPath: LPCWSTR,
+        pszEnvironment: LPCWSTR,
+    ) -> HRESULT;
+}
+ENUM!{enum EPrintPropertyType {
+    kPropertyTypeString = 1,
+    kPropertyTypeInt32,
+    kPropertyTypeInt64,
+    kPropertyTypeByte,
+    kPropertyTypeTime,
+    kPropertyTypeDevMode,
+    kPropertyTypeSD,
+    kPropertyTypeNotificationReply,
+    kPropertyTypeNotificationOptions,
+    kPropertyTypeBuffer,
+}}
+ENUM!{enum EPrintXPSJobProgress {
+    kAddingDocumentSequence = 0,
+    kDocumentSequenceAdded = 1,
+    kAddingFixedDocument = 2,
+    kFixedDocumentAdded = 3,
+    kAddingFixedPage = 4,
+    kFixedPageAdded = 5,
+    kResourceAdded = 6,
+    kFontAdded = 7,
+    kImageAdded = 8,
+    kXpsDocumentCommitted = 9,
+}}
+ENUM!{enum EPrintXPSJobOperation {
+    kJobProduction = 1,
+    kJobConsumption,
+}}
+STRUCT!{struct PrintPropertyValue_value_propertyBlob {
+    cbBuf: DWORD,
+    pBuf: LPVOID,
+}}
+UNION2!{union PrintPropertyValue_value {
+    [usize; 2],
```

Comment:
For 32-bit platforms `usize` has an alignment of 4 while `u64` has an alignment of 8. It's an important difference.